org.sonarlint.eclipse.core.tests/src/test/java/org/sonarlint/eclipse/tests/common/WorkspaceHelpers.java

```diff
@@ -26,28 +26,57 @@
 import java.nio.file.Path;
 import java.nio.file.SimpleFileVisitor;
 import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Arrays;
 import org.eclipse.core.resources.IProject;
 import org.eclipse.core.resources.IWorkspace;
 import org.eclipse.core.resources.IWorkspaceRunnable;
 import org.eclipse.core.resources.ResourcesPlugin;
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.core.runtime.Status;
+import org.sonarlint.eclipse.core.internal.SonarLintCorePlugin;
 
 /**
- * @author Evgeny Mandrikov
+ * Inspired by m2e
  */
 public final class WorkspaceHelpers {
 
-  public static void cleanWorkspace() throws CoreException, IOException {
-    doCleanWorkspace();
+  public static void cleanWorkspace() throws InterruptedException, CoreException {
+    Exception cause = null;
+    int i;
+    for (i = 0; i < 10; i++) {
+      try {
+        System.gc();
+        doCleanWorkspace();
+      } catch (InterruptedException e) {
+        throw e;
+      } catch (OperationCanceledException e) {
+        throw e;
+      } catch (Exception e) {
+        cause = e;
+        e.printStackTrace();
+        System.out.println(i);
+        Thread.sleep(6 * 1000);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS2925)